pineappl_capi/include/PineAPPL.hpp

```diff
@@ -136,21 +196,119 @@ struct Order {
   /** @brief Exponent of the electromagnetic coupling. */
   std::uint32_t alpha;
 
-  /** @brief Exponent of the logarithm of the scale factor of the renomalization
-   * scale. */
+  /** @brief Exponent of the logarithm of the scale factor of the
+   * renomalization scale. */
   std::uint32_t logxir;
 
   /** @brief Exponent of the logarithm of the scale factor of the factorization
    * scale. */
   std::uint32_t logxif;
 };
 
-/** @brief The central grid object. */
-struct Grid {
+/** @brief Extension of `Order` to accommodate for v1 representation. */
+struct OrderV1 {
```

Comment:
I would move all the `V1` objects in a `V1` namespace instead

Reply:
Or we simply nuke everything `v0`. This can still be used by using an old header.

Reply:
> Or we simply nuke everything `v0`. This can still be used by using an old header.

I like this 😄 and I'd do this actually. In this way, we won't need a complex data structure with Grid inheritance.